File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> numMap;
        int i = 0;
            for(int num : nums){
                if(numMap.find(num) != numMap.end()){
                    return true;
                }
                numMap.insert({num, i});
                i++;
            }
            return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numDict = {}
        for i, num in enumerate(nums):
            comp = target - nums[i]
            if comp in numDict:
                return sorted([i, numDict[comp]])
            numDict[nums[i]] = i
        return [-1,-1]
        